bindings/csharp/Ckzg.Bindings/Ckzg.cs

```diff
@@ -1,115 +1,231 @@
+        }
+    }
 
     /// <summary>
-    /// Given a blob and a KZG proof, verify that the blob data corresponds to the provided commitment.
+    ///     Given a blob and a KZG proof, verify that the blob data corresponds to the provided commitment.
     /// </summary>
-    /// <param name="result">True if the proof is valid</param>
     /// <param name="blob"></param>
     /// <param name="commitment"></param>
     /// <param name="proof"></param>
-    /// <param name="ts">Trusted setup settings</param>
+    /// <param name="ckzgSetup">Trusted setup settings</param>
     /// <returns></returns>
-    [DllImport("ckzg", EntryPoint = "verify_blob_kzg_proof", CallingConvention = CallingConvention.Cdecl)]
-    public unsafe static extern Ret VerifyBlobKzgProof(bool* result, byte* blob, byte* commitment_bytes, byte* proof_bytes, IntPtr ts);
+    public static unsafe bool VerifyBlobKzgProof(ReadOnlySpan<byte> blob, ReadOnlySpan<byte> commitment,
+        ReadOnlySpan<byte> proof, IntPtr ckzgSetup)
+    {
+        if (blob.Length != BytesPerBlob) throw new ArgumentException("Invalid blob size", nameof(blob));
+
+        if (commitment.Length != BytesScalar) throw new ArgumentException("Invalid z size", nameof(commitment));
+
+        if (proof.Length != BytesPerProof) throw new ArgumentException("Invalid proof size", nameof(proof));
+
+        if (ckzgSetup == IntPtr.Zero) throw new InvalidOperationException("KZG is not initialized");
+
+        fixed (byte* blobPtr = blob, commitmentPtr = commitment, proofPtr = proof)
+        {
+            KzgResult kzgResult = VerifyBlobKzgProof(out var result, blobPtr, commitmentPtr, proofPtr, ckzgSetup);
+            ThrowOnError(kzgResult);
+            return result;
+        }
+    }
 
     /// <summary>
-    /// Given a list of blobs and blob KZG proofs, verify that they correspond to the provided commitments.
+    ///     Given a list of blobs and blob KZG proofs, verify that they correspond to the provided commitments.
     /// </summary>
-    /// <param name="result">True if the proofs are valid</param>
     /// <param name="blobs">Blobs as a flattened byte array</param>
     /// <param name="commitments">Commitments as a flattened byte array</param>
     /// <param name="proofs">Proofs as a flattened byte array</param>
     /// <param name="count">The number of blobs/commitments/proofs</param>
-    /// <param name="ts">Trusted setup settings</param>
+    /// <param name="ckzgSetup">Trusted setup settings</param>
     /// <returns></returns>
-    [DllImport("ckzg", EntryPoint = "verify_blob_kzg_proof_batch", CallingConvention = CallingConvention.Cdecl)]
-    public unsafe static extern Ret VerifyBlobKzgProofBatch(bool* result, byte* blobs, byte* commitments_bytes, byte* proofs_bytes, int count, IntPtr ts);
-}
+    public static unsafe bool VerifyBlobKzgProofBatch(ReadOnlySpan<byte> blobs, ReadOnlySpan<byte> commitments,
+        ReadOnlySpan<byte> proofs, int count, IntPtr ckzgSetup)
```

Comment:
Maybe the idea is that this way the bindings don't need to go to the effort of allocating a flat array to pass it to c-kzg (like the node bindings do).

Reply:
Yep, we can probably optimize by removing splitting while deserializing from SSZ and than flatten while passing to c-kzg. Still can be two interfaces if a consumer will want to verify a batch of blobs from different transactions or like that.

Reply:
I guess the ideal approach would be to pass a list of byte arrays (so that you don't need `count`) but then be able to interpret that list as a flat array of bytes inside C so that you can pass them to verify_batch in c-kzg. You can do this in languages [like Rust](https://stackoverflow.com/a/68905410) but I don't know enough about C#.

No strong opinion from my part, as long as its convenient from the client's PoV and the code is reasonable (which I think it currently is).